tests/HTMLPurifierTest.hack

```diff
@@ -425,4 +425,82 @@ class HTMLPurifierTest extends HackTest {
 		expect($clean_html)->toEqual($expected_html);
 		echo "finished.\n\n";
 	}
+
+	public function testRel(): void {
+		echo "\nrunning testRel()...";
+		$policy = new HTMLPurifier\HTMLPurifier_Policy(
+			dict[
+				'b' => vec[],
+				'ul' => vec[],
+				'li' => vec[],
+				'ol' => vec[],
+				'h2' => vec[],
+				'h4' => vec[],
+				'br' => vec[],
+				'div' => vec[],
+				'strong' => vec[],
+				'del' => vec[],
+				'em' => vec[],
+				'pre' => vec[],
+				'code' => vec[],
+				'table' => vec[],
+				'tbody' => vec[],
+				'td' => vec[],
+				'th' => vec[],
+				'thead' => vec[],
+				'tr' => vec[],
+				'a' => vec['id', 'name', 'href', 'target', 'rel'],
+				'h3' => vec['class'],
+				'p' => vec['class'],
+				'aside' => vec['class'],
+				'img' => vec['src', 'alt', 'class', 'width', 'height', 'srcset', 'sizes'],
+			],
+		);
+		$config = HTMLPurifier\HTMLPurifier_Config::createDefault();
+		$purifier = new HTMLPurifier\HTMLPurifier($config, $policy);
+		$dirty_html = '<p><a rel="ab" name="bobcat"></a></p>';
+		$clean_html = $purifier->purify($dirty_html);
+		$expected_html = '<p><a name="bobcat"></a></p>';
+		expect($clean_html)->toEqual($expected_html);
+		echo "finished.\n\n";
+	}
+
+	public function testAtagTargetAttribute(): void {
+		echo "\nrunning testAtagTargetAttribute()...";
+		$policy = new HTMLPurifier\HTMLPurifier_Policy(
+			dict[
+				'b' => vec[],
+				'ul' => vec[],
+				'li' => vec[],
+				'ol' => vec[],
+				'h2' => vec[],
+				'h4' => vec[],
+				'br' => vec[],
+				'div' => vec[],
+				'strong' => vec[],
+				'del' => vec[],
+				'em' => vec[],
+				'pre' => vec[],
+				'code' => vec[],
+				'table' => vec[],
+				'tbody' => vec[],
+				'td' => vec[],
+				'th' => vec[],
+				'thead' => vec[],
+				'tr' => vec[],
+				'a' => vec['id', 'name', 'href', 'target', 'rel'],
+				'h3' => vec['class'],
+				'p' => vec['class'],
+				'aside' => vec['class'],
+				'img' => vec['src', 'alt', 'class', 'width', 'height', 'srcset', 'sizes'],
+			],
+		);
+		$config = HTMLPurifier\HTMLPurifier_Config::createDefault();
+		$purifier = new HTMLPurifier\HTMLPurifier($config, $policy);
+		$dirty_html = '<p><a name="bobcat" target="_blank"></a></p>';
+		$clean_html = $purifier->purify($dirty_html);
+		$expected_html = '<p><a name="bobcat" target="_blank"></a></p>';
```

Comment:
I think we expect `rel=noopener noreferer` to be added here 🤔

Reply:
Yeah, based on [this message](https://slack-pde.slack.com/archives/C014TKJC9GV/p1602627263013000?thread_ts=1602615910.010100&cid=C014TKJC9GV) from Fikrie, I think we want this to show up?

Reply:
I think it is not that with dirty_html = `<p><a name="bobcat" target="_blank"></a></p>` PHP sanitizer will add `rel="noopener noreferer"` 
The problem is with a not valid `rel` attribute value, the attribute should be discarded.

Reply:
I will post the dirty html vs HACK html vs PHP html in channel for reference.